climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+                                   hemisphere=None):
+        """ Summarize historic fires into fire seasons.
+
+        Fires are summarized by taking the max intensity at each grid point.
+
+        Parameters
+        ----------
+            year_start : int, optional
+                start year; fires before that are cut; no cut if not specified
+            year_end : int, optional
+                end year; fires after that are cut; no cut if not specified
+            hemisphere : str, optional
+                'SHS' or 'NHS' to define fire seasons
+
+        Returns
+        -------
+            self : climada.hazard.WildFire instance
+
+        """
+
+        # define hemisphere
+        if hemisphere is None:
+            if self.centroids.lat[0] > 0:
+                hemisphere = 'NHS'
+            elif self.centroids.lat[0] < 0:
+                hemisphere = 'SHS'
+
+        # define years
+        fire_years = np.array([date.fromordinal(i).year for i in self.date])
+        years = np.arange(np.min(fire_years), np.max(fire_years)+1)
+        if year_start is not None:
+            years = np.delete(years, np.argwhere(years < year_start))
+        if year_end is not None:
+            years = np.delete(years, np.argwhere(years > year_end))
+
+        # summarize to fire season
+        date_new = np.zeros(len(years), int)
+        date_end_new = np.zeros(len(years), int)
+        n_fires = np.zeros(len(years), int)
+        intensity_new = sparse.lil_matrix(np.zeros((len(years), len(self.centroids.lat))))
+
+        for i, year in enumerate(years):
+            if hemisphere == 'NHS':
+                start = date.toordinal(date(year, 1, 1))
+                end = date.toordinal(date(year+1, 1, 1))
+            elif hemisphere == 'SHS':
+                start = date.toordinal(date(year, 7, 1))
+                end = date.toordinal(date(year+1, 7, 1))
+
+            date_new[i] = start
+            date_end_new[i] = end
+            idx = np.where((self.date > start-1) & \
+                           (self.date < end + 1))
+            n_fires[i] = len(idx)
+            intensity_new[i] = sparse.csr_matrix( \
+                                    np.amax(self.intensity[idx], 0))
+
+        # save
+        self.tag = TagHazard('WFseason')
+        self.units = 'K' # Kelvin brightness
+
+        # Following values are defined for each fire season
+        self.event_id = np.arange(1, len(years)+1).astype(int)
+        self.event_name = list(map(str, years))
+        self.date = date_new
+        self.date_end = date_end_new
+        self.n_fires = n_fires
+        self.orig = np.ones(len(years), bool)
+        self._set_frequency()
+
+        # Following values are defined for each fire and centroid
+        self.intensity = intensity_new.tocsr()
+        self.fraction = self.intensity.copy()
+        self.fraction.data.fill(1.0)
+
+
+    @staticmethod
+    def _clean_firms_csv(csv_firms):
```

Comment:
A method should do ideally one thing. There is absolutely no need to have the data loading in this method. Please make this method simpler. It should just clean a FIRMS dataframe. Move the loading of the data out.

Reply:
A function which takes arguments of different types are not a good idea, they make the code unnecessarily complex. If it is only one line to read the csv, then you should simply do that outside of this function. You do not need to create a separate function for that. Just remove the data loading from `_clean_firms_csv`, call it `_clean_firms_df` and accept as input only the dataframe. Voila.

Reply:
alright - I've adjusted (cafc032a)
the whole method now requires a dataframe to start with (hence reading of the data is now done outside of the WF class)